.github/workflows/test.yml

```diff
@@ -0,0 +1,45 @@
+name: Test
+
+permissions:
+  contents: write
+
+on: [pull_request]
+
+jobs:
+  test:
+    name: Run Unit Tests and Linter
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          cache: true
+          go-version-file: 'go.mod'
+
+      - name: Cache Go Dependencies
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Install Dependencies
+        run: |
+          go mod download
+          go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.49.0
+          go mod tidy
+
+      - name: Run Tests
+        run: make test-ci
```

Comment:
I noticed we're removing the CI tests from the makefile. Do we still need this job to include the ci tests here as well then?

Reply:
So in the Makefile, we aren't exactly _removing_ the CI tests, we're instead _no longer running_ our tests on CircleCI. We still need to keep the `test-ci` Makefile command since we still want to run the tests differently in the CI (Continuous Integration) environment. By differently, I mean we need to still generate the code coverage report after each test run.
In order words, we are no longer using CircleCI as a CI environment and instead are solely using GitHub Actions as our CI environment.